app.go

```diff
@@ -98,12 +98,65 @@ type App struct {
 	ready chan struct{}
 }
 
-// NewWindow creates a window that will oad the given URL and will display
+// WindowOptions contains options for creating windows
+type WindowOptions struct {
+	URL              string // Initial URL to load; leave empty to load nothing
+	Title            string // String to display in title bar
+	Width            int    // Width in pixels. Set to zero to use OS default.
+	Height           int    // Height in pixels. Set to zero to use OS default.
+	X                int    // X offset from left in pixels. Set to zero to use OS default.
+	Y                int    // Y offset from top in pixels. Set to zero to use OS default.
+	TitleBar         bool   // Whether the window title bar
+	Frame            bool   // Whether the window has a frame
+	Resizable        bool   // Whether the window border can be dragged to change its shape
+	CloseButton      bool   // Whether the window has a close button
+	MinButton        bool   // Whether the window has a miniaturize button
+	FullScreenButton bool   // Whether the window has a full screen button
+	Menu             []MenuEntry
+}
+
+type Window struct {
+	cwindow *C.gallium_window_t
+}
+
+// OpenWindow creates a window that will load the given URL and will display
 // the given title
-func (b *App) NewWindow(url, title string) error {
-	log.Println("=== gallium.NewWindow ===")
-	cerr := newCerr()
-	defer cerr.free()
-	C.GalliumCreateWindow(C.CString(url), C.CString(title), &cerr.c)
-	return nil
+func (b *App) OpenWindow(opts ...func(*WindowOptions)) (*Window, error) {
```

Comment:
Yeah I started here too, but the problem is that it's impossibel to make the zero values for `X, Y, Width, Height` be sensible defaults. One solution was to check whether these vars are zero in `OpenWindow` and overwrite with defaults at that point, but this is tricky especially for `X` and `Y` because it should be possible to position your window explicitly at `0,0`, but that should not be the default if you do not provide any explicit `X` and `Y`. 

Another solution was to have some defaults that you can get with
```
opts := gallium.DefaultWindowOptions
opts.CloseButton = false
opts.Width = 800
opts.Height = 500
app.OpenWindow(opts)
```

but that just seems a little ugly. Although maybe not as ugly as what is here now. I'm not sure